src/docker/functions/container/exec-container.ts

```diff
@@ -27,17 +31,18 @@ export const execContainer = async (container: Dockerode.Container, command: Com
   }
 };
 
-const startExec = async (exec: Dockerode.Exec): Promise<Readable> => {
+const startExec = async (exec: Dockerode.Exec, options?: Dockerode.ExecStartOptions): Promise<Readable> => {
   try {
-    const options = {
+    const defaultOptions = {
```

Comment:
I guess we should merge `options` and `defaultOptions`, else if someone only specifies `Tty`, then the other fields wouldn't be set

Reply:
Thank you very much for your prompt response.

That’s a great idea. Inspired by how StopOptions are implemented for the stopContainer method, I made following changed:
#### interface ExecOptions is created in "test-containers.ts"
```
// from "test-containers.ts"
export interface ExecOptions {
  Tty: boolean;
  Detach: boolean;
  stdin: boolean;
}
```
- Merging of default and specific ExecOptions properties is done within "started-generic-container.ts" in a private method
- ExecContainerOptions type is declared in "exec-container" for local use -> based on StopContainerOptions from "stop-container.ts"
- execContainer in "exec-container.ts" now receives options as mandatory argument, merging is done by the caller.

## Unused properties
Some of the previously hardcoded options, such as stdout and stderr, are inexistent in the Dockerode.ExecStartOptions specification. So, I also removed these unused options in interface and type definitions.
```
// From Dockerode
interface ExecStartOptions {
        // hijack and stdin are used by docker-modem
        hijack?: boolean | undefined;
        stdin?: boolean | undefined;
        // Detach and Tty are used by Docker's API
        Detach?: boolean | undefined;
        Tty?: boolean | undefined;
        abortSignal?: AbortSignal;
}
```

## Hardcoded argument

"port-check.ts" makes a direct call to "execContainer", by-passing the merging of options that is done via "StartedGenericContainer ". 
That’s the only usage of “execContainer” function from outside of the “StartedGenericContainer” instance. 
We have 2 possibilities: 
-  keep it as is, with the hardcoded argument in "port-check.ts" 
- do the merging of options within “startExec” function, exporting the “ExecContainerOptions” for use in "started-generic-container” and removing the ExecOptions interface.